### PR TITLE
refactor: Logger parse structure

### PR DESCRIPTION
**Pull request by V3ntus**
## About

Related to #421

~~Unless a simple logging configuration is setup, log messages should be hidden.~~ Users can set the level with `logging.basicConfig(level=logging.DEBUG)` ex. if they desire. `interactions` should then propagate its loggers to that StreamHandler

A list of loggers (their names) is present in `base.Data` which gets populated after importing `interactions`

## Checklist

- [X] I've ran `pre-commit` to format and lint the change(s) made.
- [X] I've checked to make sure the change(s) work on `3.8.6` and higher.
- [ ] This fixes/solves an [Issue](https://github.com/goverfl0w/discord-interactions/issues).
  - (If existent):
- [X] I've made this pull request for/as: (check all that apply)
  - [ ] Documentation
  - [ ] Breaking change
  - [ ] New feature/enhancement
  - [X] Bugfix

### CODE REVIEW

**Comment by Catalyst4222** [comment]
Personally, I think that Warning, Error, and Critical should be shown if it directly relates to the user

So (as examples) a duplicate command should warning/error to stderr, but the critical from missing attributes shouldn't be shown

**Comment by V3ntus** [comment]
> Personally, I think that Warning, Error, and Critical should be shown if it directly relates to the user

I agree. I suppose NullHandlers would be a bad idea as default then. The missing attribute warning spam gets somewhat irritating, and confusing for some as well. Maybe we can pass errors and critical?

**Comment by V3ntus** [comment]
e500293 should allow errors, but still give users the ability to define a `logging.basicConfig` with a level of their choice. This also should prevent duplicate log messages in console. 

Tested

File: interactions/base.py [hunk 1]
```diff
@@ -18,9 +18,31 @@
 
 
 class Data:
-    """A class representing constants for the library."""
+    """A class representing constants for the library.
 
-    LOGGER: ClassVar[int] = logging.WARNING
+    :ivar LOGGERS List[str]: A list of all loggers registered from this library
```
Comment (sillyElvy) [inline interactions/base.py:24]: Add the log level to the Docstring, too

File: interactions/base.py [hunk 2]
```diff
@@ -18,9 +18,31 @@
...
+    handler: Optional[logging.Handler] = logging.StreamHandler(),
+) -> logging.Logger:
+    _logger = logging.getLogger(logger) if isinstance(logger, str) else logger
+    _logger_name = logger if isinstance(logger, str) else logger.name
+    if len(_logger.handlers) > 1:
+        _logger.removeHandler(_logger.handlers[0])
+    _handler = handler
+    _handler.setFormatter(CustomFormatter)
+    _handler.setLevel(Data.LOG_LEVEL)
+    _logger.addHandler(handler)
```
Comment (sillyElvy) [inline interactions/base.py]: add `_handler`not `handler`

**Review by sillyElvy — CHANGES_REQUESTED** [review]
Except those two smaller issues this looks fine to me

**Comment by V3ntus** [comment]
> Except those two smaller issues this looks fine to me

Thanks, missed those two

**Comment by sillyElvy** [comment]
> > Except those two smaller issues this looks fine to me
> 
> Thanks, missed those two

no problem

**Review by sillyElvy — APPROVED** [review]
LGTM 👍

**Comment by i0bs** [comment]
I'm willing to include this as a refactor instead of a feature commit, but there's a merge conflict stopping me from doing that.

**Comment by V3ntus** [comment]
Alright I'll try to get that resolved today

**Comment by V3ntus** [comment]
Merged unstable, I don't see merge conflicts anymore, but let me know if there's still something wrong

**Comment by i0bs** [comment]
Would recommend running `pre-commit` again, the checks have failed.

**Comment by i0bs** [comment]
This PR LGTM. Thank you for contributing to the library for the first time, and welcome to the contributors club. 😃

**Pull request merged**